From de8c443e0d779d8eaf5c36ebca17af8b1fb8a15b Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 10 May 2022 10:42:09 +0300
Subject: [PATCH 38/64] arm: dts: s32cc: Reposition reserved memory and
 firmware nodes

Move reserved memory and firmware nodes right after aliases.

Issue: ALB-8823
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/s32cc.dtsi | 107 +++++++++++++++++++---------------------
 arch/arm/dts/s32g3.dtsi |   8 +--
 2 files changed, 56 insertions(+), 59 deletions(-)

diff --git a/arch/arm/dts/s32cc.dtsi b/arch/arm/dts/s32cc.dtsi
index 552a3580de..cc1e662a45 100644
--- a/arch/arm/dts/s32cc.dtsi
+++ b/arch/arm/dts/s32cc.dtsi
@@ -13,6 +13,11 @@
 	#address-cells = <2>;
 	#size-cells = <2>;
 
+	chosen {
+		stdout-path = "serial0:115200n8";
+		tick-timer = &pit0;
+	};
+
 	aliases {
 		adc0 = &adc0;
 		adc1 = &adc1;
@@ -36,53 +41,6 @@
 		serial1 = &uart1;
 	};
 
-	chosen {
-		stdout-path = "serial0:115200n8";
-		tick-timer = &pit0;
-	};
-
-	cmu: cmu@4005c000 {
-		reg = <0x0 0x4005c000 0x0 0x1000>;
-		clocks = <&clks S32GEN1_SCMI_CLK_CMU_MODULE>,
-			 <&clks S32GEN1_SCMI_CLK_CMU_REG>;
-	};
-
-	cpus: cpus {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		u-boot,dm-pre-reloc;
-
-		cpu0: cpu@0 {
-			device_type = "cpu";
-			compatible = "arm,cortex-a53";
-			reg = <0x0 0x0>;
-
-			nvmem-cells = <&soc_letter>, <&part_no>,
-			    <&soc_major>, <&soc_minor>, <&soc_subminor>;
-			nvmem-cell-names = "soc_letter", "part_no",
-			    "soc_major", "soc_minor", "soc_subminor";
-			u-boot,dm-pre-reloc;
-		};
-
-		cpu1: cpu@1 {
-			device_type = "cpu";
-			compatible = "arm,cortex-a53";
-			reg = <0x0 0x1>;
-		};
-
-		cpu2: cpu@100 {
-			device_type = "cpu";
-			compatible = "arm,cortex-a53";
-			reg = <0x0 0x100>;
-		};
-
-		cpu3: cpu@101 {
-			device_type = "cpu";
-			compatible = "arm,cortex-a53";
-			reg = <0x0 0x101>;
-		};
-	};
-
 	reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <2>;
@@ -90,14 +48,14 @@
 
 		scmi_shbuf: shm@d0000000 {
 			compatible = "arm,scmi-shmem";
-			reg = <0x0 0xd0000000 0x0 0x001000>;
+			reg = <0x0 0xd0000000 0x0 0x400000>;    /* 4 MB */
 			no-map;
 		};
 
 #ifdef CONFIG_HSE_SECBOOT
-		hse_reserved: hse_reserved@0x84000000 {
+		hse_reserved: shm@84000000 {
+			reg = <0x0 0x84000000 0x0 0x80000>;	/* 512k */
 			no-map;
-			reg = <0 0x84000000 0 0x40000>; /* 256K */
 		};
 #endif
 	};
@@ -105,20 +63,17 @@
 	firmware {
 		u-boot,dm-pre-reloc;
 
-		scmi {
+		scmi: scmi {
 			u-boot,dm-pre-reloc;
 			compatible = "arm,scmi-smc";
-			arm,smc-id = <0xc20000fe>;
 			shmem = <&scmi_shbuf>;
+			arm,smc-id = <0xc20000fe>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-
 			status = "okay";
 
 			clks: protocol@14 {
 				u-boot,dm-pre-reloc;
-				#address-cells = <2>;
-				#size-cells = <2>;
 
 				reg = <0x14>;
 				#clock-cells = <1>;
@@ -131,6 +86,42 @@
 		};
 	};
 
+	cpus: cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		u-boot,dm-pre-reloc;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a53";
+			reg = <0x0>;
+
+			nvmem-cells = <&soc_letter>, <&part_no>,
+			    <&soc_major>, <&soc_minor>, <&soc_subminor>;
+			nvmem-cell-names = "soc_letter", "part_no",
+			    "soc_major", "soc_minor", "soc_subminor";
+			u-boot,dm-pre-reloc;
+		};
+
+		cpu1: cpu@1 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a53";
+			reg = <0x1>;
+		};
+
+		cpu2: cpu@100 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a53";
+			reg = <0x100>;
+		};
+
+		cpu3: cpu@101 {
+			device_type = "cpu";
+			compatible = "arm,cortex-a53";
+			reg = <0x101>;
+		};
+	};
+
 	uart0:serial@401C8000 {
 		device_type = "serial";
 		compatible = "nxp,s32cc-linflexuart";
@@ -398,6 +389,12 @@
 		status = "okay";
 	};
 
+	cmu: cmu@4005c000 {
+		reg = <0x0 0x4005c000 0x0 0x1000>;
+		clocks = <&clks S32GEN1_SCMI_CLK_CMU_MODULE>,
+			 <&clks S32GEN1_SCMI_CLK_CMU_REG>;
+	};
+
 	adc0: saradc@401f8000 {
 		compatible = "nxp,s32cc-adc";
 		reg = <0x0 0x401f8000 0x0 0x1000>;
diff --git a/arch/arm/dts/s32g3.dtsi b/arch/arm/dts/s32g3.dtsi
index ae5d8fb22c..53ca199f92 100644
--- a/arch/arm/dts/s32g3.dtsi
+++ b/arch/arm/dts/s32g3.dtsi
@@ -46,25 +46,25 @@
 	cpu4: cpu@2 {
 		device_type = "cpu";
 		compatible = "arm,cortex-a53";
-		reg = <0x0 0x2>;
+		reg = <0x2>;
 	};
 
 	cpu5: cpu@3 {
 		device_type = "cpu";
 		compatible = "arm,cortex-a53";
-		reg = <0x0 0x3>;
+		reg = <0x3>;
 	};
 
 	cpu6: cpu@102 {
 		device_type = "cpu";
 		compatible = "arm,cortex-a53";
-		reg = <0x0 0x102>;
+		reg = <0x102>;
 	};
 
 	cpu7: cpu@103 {
 		device_type = "cpu";
 		compatible = "arm,cortex-a53";
-		reg = <0x0 0x103>;
+		reg = <0x103>;
 	};
 };
 
-- 
2.17.1

